aclocal/ax_javac_and_java.m4

```diff
@@ -118,7 +118,7 @@ AC_DEFUN([AX_CHECK_JAVA_CLASS],
 AC_DEFUN([AX_CHECK_ANT_VERSION],
          [
           AC_MSG_CHECKING(for ant version > $2)
-          ANT_VALID=`expr $($1 -version 2>/dev/null | sed -n 's/.*version \(@<:@0-9\.@:>@*\).*/\1/p') \>= $2`
+          ANT_VALID=`expr "x$(printf "$2\n$($1 -version 2>/dev/null | sed -n 's/.*version \(@<:@0-9\.@:>@*\).*/\1/p')" | sort -t '.' -k 1,1 -k 2,2 -k 3,3 -g | sed -n 1p)" = "x$2"`
```

Comment:
I don't see this working on my bash shell:

```
jking@ubuntu:~/thrift/github/thrift$ ant -version
Apache Ant(TM) version 1.9.8 compiled on January 19 2017

jking@ubuntu:~/thrift/github/thrift$ ant -version | sed -n 's/.*version \(@<:@0-9\.@:>@*\).*/\1/p'
(no output)
```

Try this instead:
```
jking@ubuntu:~/thrift/github/thrift$ ant -version | cut -d' ' -f4
1.9.8
```

Test:

If I require 1.9.9, fails, if I require 1.9.7, passes:
```
jking@ubuntu:~/thrift/github/thrift$ expr "x$(printf "1.9.9\n$(ant -version | cut -d' ' -f4)" | sort -t '.' -k 1,1 -k 2,2 -k 3,3 -g | sed -n 1p)" = "x1.9.9"
0
jking@ubuntu:~/thrift/github/thrift$ expr "x$(printf "1.9.7\n$(ant -version | cut -d' ' -f4)" | sort -t '.' -k 1,1 -k 2,2 -k 3,3 -g | sed -n 1p)" = "x1.9.7"
1
```

Reply:
The issue that the pull request is fixing is the version comparison part. The ant version extraction is working like charm and doesn't need to be changed.

@jeking3, you are not testing accurately. Thats not a bash script file, this is a Autoconf/m4 macro and the following:
`ant -version | sed -n 's/.*version \(@<:@0-9\.@:>@*\).*/\1/p'`
 at the time of running the bash command will have the 'Quadrigraphs'(@<:@) escaped:
`ant -version | sed -n 's/.*version \([0-9\.]*\).*/\1/p'`

For further details check the docs: 
https://www.gnu.org/software/autoconf/manual/autoconf.html#Quadrigraphs

I hope this clears the things a bit.
Thanks